modules/sdk-coin-apt/src/lib/transactionBuilder/fungibleAssetTransactionBuilder.ts

```diff
@@ -0,0 +1,56 @@
+import { TransactionBuilder } from './transactionBuilder';
+import { BaseCoin as CoinConfig } from '@bitgo/statics';
+import { FungibleAssetTransaction } from '../transaction/fungibleAssetTransaction';
+import { TransactionType } from '@bitgo/sdk-core';
+import BigNumber from 'bignumber.js';
+import utils from '../utils';
+import { TransactionPayload, TransactionPayloadEntryFunction } from '@aptos-labs/ts-sdk';
+import { FUNGIBLE_ASSET } from '../constants';
+
+export class FungibleAssetTransactionBuilder extends TransactionBuilder {
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+    this._transaction = new FungibleAssetTransaction(_coinConfig);
+  }
+
+  protected get transactionType(): TransactionType {
+    return TransactionType.SendToken;
+  }
+
+  /** @inheritdoc */
+  validateTransaction(transaction?: FungibleAssetTransaction): void {
+    if (!transaction) {
+      throw new Error('transaction not defined');
+    }
+    this.validateAddress({ address: transaction.sender });
+    this.validateAddress({ address: transaction.recipient.address });
+    this.validateValue(new BigNumber(transaction.recipient.amount));
+    this.validateAddress({ address: transaction.assetId });
```

Comment:
Move the common portion to the parent class wherever possible.
```suggestion
    super.validateTransaction(transaction);
    this.validateAddress({ address: transaction.assetId });
```